### PR TITLE
add smart heroku-style root redirect

### PR DESCRIPTION
**Pull request by LegoStormtroopr**

### CODE REVIEW

File: python/aristotle-metadata-registry/aristotle_mdr/urls/aristotle.py [hunk 1]
```diff
@@ -18,7 +18,11 @@
 
 
 urlpatterns=[
-    url(r'^$', TemplateView.as_view(template_name='aristotle_mdr/static/home.html'), name="home"),
+    url(r'^$', views.SmartRoot.as_view(
+        unauthenticated_pattern='aristotle_mdr:home',
+        authenticated_pattern="aristotle_mdr:userHome"
```
Comment (derrix060) [inline python/aristotle-metadata-registry/aristotle_mdr/urls/aristotle.py]: don't mix `"` and `'`.

From [PEP8](https://www.python.org/dev/peps/pep-0008/#id15):
```
...code is read much more often than it is written.
...
A style guide is about consistency. Consistency with this style guide is important. Consistency within a project is more important. Consistency within one module or function is the most important.
```

**Comment by LegoStormtroopr** [comment]
@harrywhite4 can you sanity check this docker setup and readme before I pull this in?

**Comment by harrywhite4** [comment]
No worries

**Comment by harrywhite4** [comment]
Looks pretty good, couple things though

The Pipfile probably wont be needed once #815 is pulled in since it could use that pipfile
Also pinning django-notifications to 1.3 will fail on the current develop branch since i pulled in a change to the template tags using the 1.4 syntax

In the docs we could suggest using --settings instead of settings an env var like this
``django-admin test aristotle_mdr.tests --settings=aristotle_mdr.tests.settings.settings``
This means you wouldn't need to run it though bash

**Comment by harrywhite4** [comment]
So does it work with DEBUG=False still, since we aren't collecting static in the Dockerfile?

**Comment by LegoStormtroopr** [comment]
I'm updating it based on those recommendations.

It probably won't work with debug off as it uses staticfiles to server the static files. This prevents the long collectstatic step in development.

I'll see what options we have with that before I pull it in

**Comment by LegoStormtroopr** [comment]
Ok - so now it has some whitenoise for static files and works with DEBUG on and off.

The README is pretty comprehensive, and development on this should be much easier now. Have another look, and if this is good I finish on pulling this in.

**Pull request merged**